### PR TITLE
Fix definition of serializer attributes with multiple calls to `attri…

### LINKED ISSUE #1091: Response with multiple lists
Hi,

is there a way to specify something like this:

render json: {projects: @projects, each_serializer: ProjectSerializer, clients: clients, each_serializer: ProjectSerializer}, status: :ok

### PR DESCRIPTION
**Pull request by beauby**
…bute` instead of one single call to `attributes`.

Fixes #1094.

### CODE REVIEW

File: lib/active_model/serializer.rb [hunk 1]
```diff
@@ -70,7 +70,7 @@ def self.attribute(attr, options = {})
       ActiveModelSerializers.silence_warnings do
         define_method key do
           object.read_attribute_for_serialization(attr)
-        end unless respond_to?(key, false) || _fragmented.respond_to?(attr)
+        end unless (key != :id && method_defined?(key)) || _fragmented.respond_to?(attr)
```
Comment (bf4) [inline lib/active_model/serializer.rb]: looks like this should be extracted to a helper..

**Comment by bf4** [comment]
:+1:

**Comment by whatthewhat** [comment]
:+1:

File: lib/active_model/serializer.rb (hunk 1 above)
Comment (beauby) [inline lib/active_model/serializer.rb]: While we're at it, should `_fragmented.respond_to?(attr)` not be `_fragmented.respond_to?(key)`? I haven't delved into the caching yet.

**Comment by joaomdmoura** [comment]
Nice @beauby, I'm merging it, Tks! :smile:

**Pull request merged**